jwst/regtest/test_lrs_slit_spec2.py

```diff
@@ -0,0 +1,51 @@
+import os
+import pytest
+from astropy.io.fits.diff import FITSDiff
+
+from jwst.pipeline.collect_pipeline_cfgs import collect_pipeline_cfgs
+from jwst.stpipe import Step
+
+@pytest.fixture(scope="module")
+def run_pipeline(jail, rtdata_module):
+    """Run the calwebb_spec2 pipeline on an ASN of nodded MIRI LRS
+       fixedslit exposures."""
+
+    rtdata = rtdata_module
+
+    # Get the cfg files
+    collect_pipeline_cfgs("config")
+
+    # Get the spec2 ASN and its members
+    rtdata.get_asn("miri/lrs/jw00623-o032_20191210t195246_spec2_001_asn.json")
+
+    # Run the calwebb_spec2 pipeline; save results from intermediate steps
+    # NOTE: THE RESAMPLE_SPEC STEP IS SKIPPED FOR NOW, BECAUSE IT HAS A BUG
+    # (the s2d image is all zeros)
+    args = ["config/calwebb_spec2.cfg", rtdata.input,
+            "--steps.resample_spec.skip=true",       # remove when bug fixed
+            "--save_bsub=true",
+            "--steps.flat_field.save_results=true",
+            "--steps.srctype.save_results=true"]
+    Step.from_cmdline(args)
+
+    return rtdata
+
+
+@pytest.mark.bigdata
+@pytest.mark.parametrize("output",[
+    "bsub", "flat_field", "srctype", "cal", "x1d"])
+def test_miri_lrs_slit_spec2(run_pipeline, fitsdiff_default_kwargs, output):
+    """Regression test of the calwebb_spec2 pipeline on MIRI
+       LRS fixedslit data using along-slit-nod pattern for
+       background subtraction."""
+
+    # Run the pipeline and retrieve outputs
+    rtdata = run_pipeline
+    rtdata.output = "jw00623032001_03102_00001_mirimage_" + output + ".fits"
+
+    # Get the truth files
+    rtdata.get_truth(os.path.join("truth", "test_miri_lrs_slit_spec2", rtdata.output))
```

Comment:
I think this will fail because `rtdata.output` is an absolute local path.  I think the way you had done it before is preferable.

Reply:
That's very interesting!  So if I assign a string to `rtdata.output`, the value of `rtdata.output` will not actually be the string that I assigned to it?

Reply:
The new way worked when I ran the test locally.

Reply:
Yes.  I suspect this is unexpected?  This is a data class after all, not just random attributes.

The reason it works here for you Howard is that `rtdata.get_truth()` is copying the output of your pipeline that the test just ran into the `truth` subdir and then comparing to output, which of course is the file that just got copied. So this is completely bypassing Artifactory.  I.e. there's no way this can fail.  If you print out the values:
```
    # Get the truth files
    print(os.path.join("truth", "test_miri_lrs_slit_spec2", rtdata.output))
    rtdata.get_truth(os.path.join("truth", "test_miri_lrs_slit_spec2", rtdata.output))
    print(rtdata.truth)
    assert 0
```
You get
```
/Users/jdavies/scratch/test_lrs_slit_spec2_run_pipeline0/jw00623032001_03102_00001_mirimage_cal.fits
/Users/jdavies/scratch/test_lrs_slit_spec2_run_pipeline0/truth/jw00623032001_03102_00001_mirimage_cal.fits
```

Reply:
Fixed in #4378